turbinia/workers/__init__.py

```diff
@@ -214,24 +214,28 @@ def __init__(self, name=None, base_output_dir=None, request_id=None):
     self.stub = None
 
   def execute(self, cmd, result, save_files=None, new_evidence=None,
-              close=False):
+              close=False, shell=False):
     """Executes a given binary and saves output.
 
     Args:
-      cmd (list): Command arguments to run
+      cmd (list|string): Command arguments to run
       result (TurbiniaTaskResult): The result object to put data into.
       save_files (list): A list of files to save (files referenced by Evidence
           objects are automatically saved, so no need to include them).
       new_evidence (list): These are new evidence objects created by the task.
           If the task is successful, they will be added to the result.
       close (bool): Whether to close out the result.
+      shell (bool): Whether the cmd is in the form of a string or a list.
 
     Returns:
       Tuple of the return code, and the TurbiniaTaskResult object
     """
     save_files = save_files if save_files else []
     new_evidence = new_evidence if new_evidence else []
-    proc = subprocess.Popen(cmd)
+    if shell:
```

Comment:
Why use Popen here and not subprocess.check_output?

```e.g: subprocess.check_output(cmd, shell=True)```

This will raise CalledProcessError on any error which we can catch and communicate upstream.

Reply:
As discussed offline, I think this captures everything we need to for now (stdin/stderr/ret code).  I'll keep it for now then.